src/System.Net.Security/tests/FunctionalTests/CertificateValidationClientServer.cs

```diff
@@ -70,11 +70,7 @@ public void Dispose()
                 Task clientConnect = clientConnection.ConnectAsync(serverEndPoint.Address, serverEndPoint.Port);
                 Task<TcpClient> serverAccept = server.AcceptTcpClientAsync();
 
-                Assert.True(
-                    Task.WaitAll(
-                        new Task[] { clientConnect, serverAccept },
-                        TestConfiguration.PassingTestTimeoutMilliseconds),
-                    "Client/Server TCP Connect timed out.");
+                Assert.True(await TestConfiguration.WhenAllOrAnyFailedWithTimeout(clientConnect, serverAccept), "Client/Server TCP Connect timed out.");
```

Comment:
Why do we want to assert rather than just letting the TimeoutException propagate?

Reply:
TBH I don't that was what was there, if you are happy for me to change to just throwing the timeout all the way through I will happily make that change tonight?

Reply:
> TBH I don't that was what was there

Presumably because Task.WaitAll returns false on timeout.

Reply:
Okay I have ripped out all of the 

Assert.True(await DoHandshake
Assert.True(await Whenxxxx

Basically I think they don't really add anything, if you throw a TimeoutException tells you what you need to know and the call stack is pretty clear (I simulated a couple), you can trace back the line in the test that caused the timeout so it didn't really add anything.

There were two tests that only had an assert on those but I switched out for assertions that told us something else about the tested code (for authentication I asserted IsAuthenticated and for Read/Write async I asserted we got some bytes not zero bytes back on the read).

Its pretty close to how I would like the tests now if they were mine :)